### DISCUSSION TITLE
How to extract cells name and formula in a space?

### DISCUSSION THREAD

**Discussion by bakerwy** [fumitoh/modelx#91] [Q&A] [answered]
Hi, 

I am now a coding heavy models with lots of variables, tables in modelx.
This work is not by one person. Some of co-workers are working on together.

As model is getting complex, we realized that it is very hard to compare cells which modified or newly coded.

so, I am looking for the method to compare between codes as Fumito the founder of modelx did as following.

https://www.diffchecker.com/rKGSoxPL/

I think first things to do is extract cells and formulas coded.
I found a way listing up cells by using **Space.cells** but it is very hard to join this with formulas.

Is there simple way to extract cells names with formulas?

**Comment by fumitoh** [comment, marked as answer]
@bakerwy Whan you save your model by the `Model.write` method or the `modelx.write_model` function,
your model is saved in a folder.  Each of the sub folders in it represents a Space in the model.
In the subfolder, `__init__.py`  includes the cells formulas as Python functions.

Below is an example of a model included in lifelib.

https://github.com/lifelib-dev/lifelib/blob/v0.9.3/lifelib/libraries/basiclife/BasicTerm_S/Projection/__init__.py#L194-L765

**Reply by bakerwy** [reply]
Oh, it is very useful. Thanks alot!
